Code/Framework/AzNetworking/AzNetworking/UdpTransport/UdpSocket.cpp

```diff
@@ -79,17 +79,15 @@ namespace AzNetworking
             {
                 const int32_t error = GetLastNetworkError();
                 AZLOG_ERROR("Failed to bind UDP socket to port %u (%d:%s)", uint32_t(port), error, GetNetworkErrorDesc(error));
+                Close();
                 return false;
             }
         }
 
-        if (!SetSocketBufferSizes(m_socketFd, net_UdpSendBufferSize, net_UdpRecvBufferSize))
-        {
-            return false;
-        }
-
-        if (!SetSocketNonBlocking(m_socketFd))
+        if (!SetSocketBufferSizes(m_socketFd, net_UdpSendBufferSize, net_UdpRecvBufferSize)
+         || !SetSocketNonBlocking(m_socketFd))
         {
+            Close();
```

Comment:
Looks like we'll need to update Socket unit tests to check for this sort of thing

Reply:
Yup, caught this when I noticed that calling Open for a used port returned a socket that was neither open nor closed, so trying to call open on it again with a new port number would actually fail. Now if open fails, you can call open again with different parameters and it will succeed.